pkg/node/node.go

```diff
@@ -269,6 +286,18 @@ func createAPIServer(cfg NodeConfig, userKey *baccrypto.UserKey) (*publicapi.Ser
 
 	safePortNumber := uint16(givenPortNumber)
 
+	var chosenAuthorizer authz.Authorizer
+
+	if len(cfg.BacalhauConfig.API.Auth.Users) > 0 || cfg.BacalhauConfig.API.Auth.Oauth2.ProviderID != "" {
+		// If new auth configuration detected, use new authorizer
+		chosenAuthorizer, err = authz.NewEntryPointAuthorizer(ctx, cfg.NodeID, cfg.BacalhauConfig.API.Auth)
+		if err != nil {
+			return nil, fmt.Errorf("error initializing users: %s", err.Error())
+		}
+	} else {
+		chosenAuthorizer = authz.NewPolicyAuthorizer(authzPolicy, userKey.PublicKey(), cfg.NodeID)
+	}
+
```

Comment:
what is the differnce between this and the `NewStandardAuthenticatorsFactory` in the factories file?

Reply:
Todo